### ISSUE TITLE
Pass header to upstream

### ISSUE THREAD

**Issue by yaslama** [h2o/h2o#565]
Hi,
I didn't find how to pass header to upstream like proxy_set_header in nginx (or fastcgi_param). Did I miss something ?

**Comment by kazuho** [comment]
There is no such setting, and I believe there is generally no need to so such thing.

You can modify the value sent by `host` header using [proxy.preserve-host](https://h2o.examp1e.net/configure/proxy_directives.html#proxy.preserve-host), disable persistent connections using [proxy.timeout.keepalive](https://h2o.examp1e.net/configure/proxy_directives.html#proxy.timeout.keepalive).

Or do you want to modify other headers?  If so, can you name them and explain why?

**Comment by yaslama** [comment]
Thanks for your answer. In fact my use case is a multi-site php installation using hhvm running in proxygen mode (the integrated web server). There is a mapping between hostnames and directories: host1->dir1, host2->dir2, etc..
I prefer to have a single place where this mapping is stored and it needs to be in the h2o config because h2o is serving the static files.
When using nginx, I use for instance the following configuration snippet:
```
server {
  listen <ip>:443 ssl http2;
  server_name <host1>;
  root <dir1>;
  set $site_directory <dir1>;
  location / {
    try_files $uri @rewrite;
  }
  location @rewrite {
    rewrite ^ /index.php;
  }
  location ~ \.php$ {
   proxy_set_header Site-Directory $site_directory;
   proxy_pass http://localhost:8000/;
  }
}
```
And in my main index.php, I use the header "Site-Directory" to know what is the root dir of the current site, and I don't need to store in another place the map between host_i and dir_i.
I hope my explanation is clear enough.
Thanks a lot!

**Comment by kazuho** [comment]
@yaslama Thank you for the clarification.  Understood.

Will consider how I should provide a way to tweak the request headers sent to backend.

**Comment by kazuho** [comment]
As covered in the test implemented in #666, it is now possible to use mruby handler's `http_request` method to issue an HTTP request with arbitrary headers to upstream, and send the response downstream.

https://github.com/h2o/h2o/blob/94a4aa2/t/50mruby-http-request.t#L47-L60 would be a good place to look at how it can be done.

Please note that the API is subject to change until we release 1.7.0.  Also, the documentation will be available at time of the release.
